WordPress/gutenberg
# [Site Editor]: Consolidate save button functionality

## What?
Fixes: https://github.com/WordPress/gutenberg/issues/60058

Currently we have many `save` buttons and some functionality is duplicated and adds complexity to the code. The save button in navigation sidebar sometimes uses the save button of document header and sometimes a custom one. When using the custom one, we almost duplicate the saving of the entities callback from the `save` button in `EntitiesSavedStates` panel.

This PR refactors the code:
1. Sidebar navigation button uses the same save button as in document header. I've added a prop whether to show the `review x item..` message, but we could also consolidate as well at some point(or here).
2. The save button in document sidebar uses the same callback to save entities with `EntitiesSavedStates` panel

The goal is to try to consolidate related components that will also help to the unification of publishing flows between post and site editors([issue](https://github.com/WordPress/gutenberg/issues/59878)).

## Testing Instructions
1. Everything should work exactly as before with no regressions
2. In site editor if we have only made changes to the current entity, we don't open the save entities panel(fix of linked issue)
